harvester_e2e_tests/integration/test_hosts.py

```diff
@@ -67,3 +127,156 @@ def test_host_poweron_state(api_client, host_state, wait_timeout):
+
+        cpus = node.get('status', {}).get('capacity', {}).get('cpu', 0)
+        assert 0 != cpus, (
+            f"Node {node_name} cpu should not be zero"
+        )
+
+        mems = node.get('status', {}).get('capacity', {}).get('memory', "0Ki")
+        mems = "".join(c for c in mems if c.isdigit())
+        mems = mems or "0"
+        assert 0 != int(mems, 10), (
+            f"Node {node_name} memory should not be zero"
+        )
+
+
+@pytest.mark.hosts
+@pytest.mark.p0
+class TestHostTriggerVmMigrate:
+    @pytest.mark.dependency()
+    def test_prerequisite(self, api_client):
+        status_code, nodes_info = api_client.hosts.get()
+
+        assert status_code == 200, f"Failed to list nodes with error: {nodes_info}"
+        assert len(nodes_info['data']) > 1, "Please use multiple hosts to test vm migration"
+
+        for node_info in nodes_info.get('data', []):
+            for condition in node_info.get('status', {}).get('conditions', []):
+                if condition.get('type', "") == "Ready":
+                    assert "True" == condition.get('status', ""), (
+                        f"Node {node_info['metadata']['name']} is not ready"
+                    )
+            assert False is node_info.get('spec', {}).get('unschedulable', False), (
+                f"Node {node_info['metadata']['name']} is unschedulable"
+            )
+
+    @pytest.mark.dependency(depends=["TestHostTriggerVmMigrate::test_prerequisite"])
+    def test_maintenance_mode(self, api_client, focal_vm, wait_timeout):
+        vm_name = focal_vm['metadata']['name']
+        vm_namespace = focal_vm['metadata']['namespace']
+        code, data = api_client.vmis.get(name=vm_name, namespace=vm_namespace)
+        assert 200 == code, (
+            f"Can't get VMI {vm_namespace}/{vm_name} with error: {code}, {data}"
+        )
+        src_host = data['status']['nodeName']
+
+        code, data = api_client.hosts.maintenance_mode(src_host, enable=True)
+        assert 200 == code, (
+            f"Failed to enable maintenance mode on node {src_host} with error: {code}, {data}",
+        )
+
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.vmis.get(name=vm_name, namespace=vm_namespace)
+            if data.get('status', {}).get('phase', "") == "Running":
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"VM {vm_namespace}/{vm_name} can't be Running with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
+
+        # teardown
+        code, data = api_client.hosts.maintenance_mode(src_host, enable=False)
+        assert 200 == code, (
+            f"Failed to disable maintenance mode on node {src_host} with error: {code}, {data}",
+        )
+
+    @pytest.mark.dependency(depends=["TestHostTriggerVmMigrate::test_maintenance_mode"])
```

Comment:
As the test will not use `maintenance_mode`, so we don't need to mark this depends on it.
But we will need make this depends on `power_off_state` and `power_on_state` to let the test be skipped if scripts is not work in previous two tests.